controllers/servicedefaults_controller.go

```diff
@@ -29,21 +30,45 @@ type ServiceDefaultsReconciler struct {
 	Log          logr.Logger
 	Scheme       *runtime.Scheme
 	ConsulClient *capi.Client
+
+	// EnableConsulNamespaces indicates that a user is running Consul Enterprise
+	// with version 1.7+ which supports namespaces.
+	EnableConsulNamespaces bool
+
+	// ConsulDestinationNamespace is the name of the Consul namespace to create
+	// all config entries in. If EnableNSMirroring is true this is ignored.
+	ConsulDestinationNamespace string
+
+	// EnableNSMirroring causes Consul namespaces to be created to match the
+	// k8s namespace of any config entry custom resource. Config entries will
+	// be created in the matching Consul namespace.
+	EnableNSMirroring bool
+
+	// NSMirroringPrefix is an optional prefix that can be added to the Consul
+	// namespaces created while mirroring. For example, if it is set to "k8s-",
+	// then the k8s `default` namespace will be mirrored in Consul's
+	// `k8s-default` namespace.
+	NSMirroringPrefix string
+
+	// CrossNSACLPolicy is the name of the ACL policy to attach to
+	// any created Consul namespaces to allow cross namespace service discovery.
+	// Only necessary if ACLs are enabled.
+	CrossNSACLPolicy string
 }
 
 // +kubebuilder:rbac:groups=consul.hashicorp.com,resources=servicedefaults,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=consul.hashicorp.com,resources=servicedefaults/status,verbs=get;update;patch
 
 func (r *ServiceDefaultsReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
-	logger := r.Log.WithValues("servicedefaults", req.NamespacedName)
+	logger := r.Log.WithValues("controller", "servicedefaults", "request", req.NamespacedName)
 	var svcDefaults consulv1alpha1.ServiceDefaults
 
 	err := r.Get(ctx, req.NamespacedName, &svcDefaults)
 	if k8serr.IsNotFound(err) {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	} else if err != nil {
-		logger.Error(err, "failed to retrieve resource")
+		logger.Error(err, "retrieving resource")
```

Comment:
So our errors don't stutter, eg. `ERROR error creating foo: failed reading file: permissions not valid` => `ERROR creating foo: reading file: permissions not valid`